### PR TITLE
Feature/ab2d 3281 add endpoints and swagger docs

### PR DESCRIPTION
**Pull request by sharshar-sb**
**JIRA Tickets:**

[AB2D-3281](https://jira.cms.gov/browse/AB2D-3281) - Create API and documentation for V2 version of R4 FHIR spec.

### What Does This PR Do?

Adds but hides the FHIR R4 V2 endpoints. 

### Usage/Deployment Instructions

Another environment variable **AB2D_V2_ENABLED** is added. By default, it's false but can be set to true to do things like test the e2e tests and eventually, turn on the endpoint.

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [X] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [ ] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [ ] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [ ] Code checked for PHI/PII exposure

### CODE REVIEW

File: api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    public static final String JOB_COMPLETE = "The job is complete.";
+    public static final String STATUS_DELAY = "A delay time in seconds before another status request will be accepted.";
+    public static final String PROGRESS = "Completion percentage, such as 50%";
+    public static final String FILE_EXPIRES = "Indicates when (an HTTP-date timestamp) the files " +
+            "listed will no longer be available for access.";
+    public static final String CANCEL = "Cancel Export Job";
+    public static final String AUTH = "Authorization";
+    public static final String CAP_DESC = "Returns the FHIR capability statement";
+    public static final String CAP_REQ = "A request for the FHIR capability statement";
+    public static final String JSON = "application/json";
```
Comment (nflearl) [inline api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java:27]: Too generic, How about APP_JSON_TYPE?

File: api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public static final String CONTENT_TYPE_DESC = "Content-Type header that matches the file format being delivered: ";
+    public static final String BULK_DNLD = "Bulk Data File Download API";
+    public static final String BULK_DNLD_DSC = "After creating a job, the API to download the generated bulk download files";
+    public static final String NOT_FOUND = "Job or file not found. ";
+    public static final String STILL_RUNNING = "The job is still in progress.";
+    public static final String STATUS_DES = "Returns a status of an export job.";
+    public static final String JOB_ID = "A job identifier";
+    public static final String FILE_NAME = "A file name";
+    public static final String DNLD_DESC = "Returns the requested file as " + NDJSON_FIRE_CONTENT_TYPE + " or " + ZIPFORMAT;
+    public static final String STATUS = "Status of export job";
```
Comment (nflearl) [inline api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java:40]: Too Generic.  How about EXPORT_JOB_STATUS?

File: api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    public static final String JOB_NOT_FOUND = "Job not found. " + GENERIC_FHIR_ERR_MSG;
+    public static final String JOB_CANCELLED_MSG = "Job canceled";
+    public static final String CAP_STMT = "FHIR capability statement";
+    public static final String CAP_API = "Provides the standard required capability statement";
+    public static final String CAP_RET = "FHIR Capability Statement Returned";
+    public static final String CONTRACT_NO = "A contract number";
+    public static final String EXPORT = "Export Claim Data";
```
Comment (nflearl) [inline api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java:47]: Too Generic, how about EXPORT_CLAIM.

File: api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    public static final String CONT_LOC = "Content-Location";
```
Comment (nflearl) [inline api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java:48]: There are definitions for Standard headers in libraries we pull in.  Both Java EE6 and Spring have HTTPHeaders.

File: api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    public static final String FILE_EXPIRES = "Indicates when (an HTTP-date timestamp) the files " +
+            "listed will no longer be available for access.";
+    public static final String CANCEL = "Cancel Export Job";
+    public static final String AUTH = "Authorization";
+    public static final String CAP_DESC = "Returns the FHIR capability statement";
+    public static final String CAP_REQ = "A request for the FHIR capability statement";
+    public static final String JSON = "application/json";
+    public static final String DOWNLOAD = "Downloads Export File";
+    public static final String DOWNLOAD_DESC = "Downloads a file produced by an export job.";
+    public static final String CONTENT_TYPE = "Content-Type";
```
Comment (nflearl) [inline api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java:30]: Use HTTPHeaders.

File: api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    public static final String STATUS = "Status of export job";
+    public static final String JOB_NOT_FOUND = "Job not found. " + GENERIC_FHIR_ERR_MSG;
+    public static final String JOB_CANCELLED_MSG = "Job canceled";
+    public static final String CAP_STMT = "FHIR capability statement";
+    public static final String CAP_API = "Provides the standard required capability statement";
+    public static final String CAP_RET = "FHIR Capability Statement Returned";
+    public static final String CONTRACT_NO = "A contract number";
+    public static final String EXPORT = "Export Claim Data";
+    public static final String CONT_LOC = "Content-Location";
+    public static final String ASYNC = "respond-async";
```
Comment (nflearl) [inline api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java:49]: Might be a standard constant for this one in HttpHeaders.

If not, how about RESPOND_ASYNC?

File: api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    public static final String OUT_FORMAT = "_outputFormat";
+    public static final String SINCE = "_since";
+    public static final String TYPE = "_type";
```
Comment (nflearl) [inline api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java:52]: Way too generic, don't even have an idea of what to suggest.

File: api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    public static final String PREFER = "Prefer";
+    public static final String EXPIRES = "Expires";
```
Comment (nflearl) [inline api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java:54]: Standard header?

File: api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    public static final String X_PROG = "X-Progress";
+    public static final String RETRY = "Retry-After";
```
Comment (nflearl) [inline api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java:56]: Standard Header?

**Review by nflearl — CHANGES_REQUESTED** [review]
Minor constant cleanup requested.

File: api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    public static final String EXPORT_STARTED = "Export request has started";
+    public static final String JOB_COMPLETE = "The job is complete.";
+    public static final String STATUS_DELAY = "A delay time in seconds before another status request will be accepted.";
+    public static final String PROGRESS = "Completion percentage, such as 50%";
+    public static final String FILE_EXPIRES = "Indicates when (an HTTP-date timestamp) the files " +
+            "listed will no longer be available for access.";
+    public static final String CANCEL_JOB = "Cancel Export Job";
+    public static final String CAP_DESC = "Returns the FHIR capability statement";
+    public static final String CAP_REQ = "A request for the FHIR capability statement";
+    public static final String APPLICATION_JSON = "application/json";
```
Comment (nflearl) [inline api/src/main/java/gov/cms/ab2d/api/controller/common/ApiText.java:25]: MediaType.APPLICATION_JSON could be used instead.

**Review by nflearl — COMMENTED** [review]
No other obvious issues.

**Review by nflearl — APPROVED** [review]
No other obvious issues.

**Review by sb-wnyffenegger — APPROVED** [review]

**Pull request merged**